odxtools/diagcodedtypes.py

```diff
@@ -514,7 +514,7 @@ def length_key(self) -> "LengthKeyParameter":
 
     def convert_internal_to_bytes(self, internal_value, encode_state: EncodeState,
                                   bit_position: int) -> bytes:
-        bit_length = encode_state.length_keys.get(self.length_key.odx_id, None)
+        bit_length = encode_state.length_keys.get(self.length_key.short_name, None)
```

Comment:
sure about using short name instead of oid?

Reply:
I'm sure that the name of a parameter is unique within the context of a given (sub-)structure, and I prefer to use the short names as indices because they are plain strings and thus quite a bit more intuitive to work with IMO...